components/post_markdown/post_markdown.tsx

```diff
@@ -57,6 +58,15 @@ export default class PostMarkdown extends React.PureComponent<Props> {
         options: {},
     };
 
+    getOptions = memoize(
```

Comment:
It feels weird to have to pre-emptively memoize the parameters passed to `Markdown`, instead of having that component own that logic itself. Should we implement a custom `shouldComponentUpdate` to replace the default one provided by the `PureComponent` that `Markdown` extends today?

Reply:
It is a little odd but I'm hesitant to introduce manual `shouldComponentUpdate` functions unless it's absolutely necessary. They'll run a lot more often than `render` will and can be easy to screw up. I might be overreacting a bit but I remember in the past overusing `shouldComponentUpdate` introduced a lot of headaches. Plus manual `shouldComponentUpdate` functions means we need to check all the props manually ourself.

Maybe this is something we should talk about and come to a guideline or heuristic on for handling these types of things.